pkg/tsdb/elasticsearch/healthcheck.go

```diff
@@ -90,31 +88,131 @@ func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthReque
 		logger.Error("Error reading response body bytes", "error", err)
 		return &backend.CheckHealthResult{
 			Status:  backend.HealthStatusUnknown,
-			Message: "Failed to read response",
-		}, err
+			Message: "Health check failed: Failed to read response",
+		}, nil
 	}
 
 	jsonData := map[string]any{}
 
 	err = json.Unmarshal(body, &jsonData)
 	if err != nil {
-		logger.Error("Error during json unmarshal of the body", "error", err)
+		truncatedBody := string(body)
+		if len(truncatedBody) > 100 {
+			truncatedBody = truncatedBody[:200] + "..."
```

Comment:
This should be `> 200`. It would be good to pull out the number into a const, maybe something like `errorBodyTruncationThreshold` or something like that.